Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -1,103 +1,123 @@
+            float anglePlayerNpc = NpcManager.LookWhereYouAreGoing(velocity);
 
-        Vector3 steering = desiredVelocity - currentVelocity;
+            //If the player is out FOV, NPC stop moving and rotate until the player is back in FOV
+            if (anglePlayerNpc > NpcGlobalVariables.FieldOfView &&
+                NpcManager.GetDistanceWithPlayer() > NpcGlobalVariables.ChasePlayerRange)
+            {
+                velocity = Vector3.zero;
+            }
 
-        Vector3 velocity = currentVelocity + steering;
+            //The NPC always walk forward.
+            NpcManager.SetAnimatorVelocity(Vector3.forward * velocity.magnitude);
+        }
 
-        if (Vector3.Magnitude(velocity) > _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity))
+        private Vector3 ComputeChaseVelocity()
         {
-            velocity = (velocity / Vector3.Magnitude(velocity)) *
-                       _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
-        }
+            
+            ComputeCurrentTargetLocation();
 
-        velocity.y = 0;
+            Vector3 desiredVelocity = Vector3.Normalize(_currentTargetLocation - NpcManager.transform.position) *
+                                      NpcManager.GetModelSpeed(NpcGlobalVariables.ChaseAcceleration);
 
-        return velocity;
-    }
+            Vector3 currentVelocity = NpcManager.Rigidbody.velocity;
 
-    private void ComputeAStar()
-    {
-        Vector3 nodeWithoutY = Vector3.zero;
-        if (_currentTargetLocation != NPCsGlobalVariables.DefaultInitialVector)
-        {
-            nodeWithoutY = new Vector3(_currentTargetLocation.x, 0, _currentTargetLocation.z);
+            Vector3 steering = desiredVelocity - currentVelocity;
+
+            var velocity = ComputeVelocity(currentVelocity, steering);
+
+            return velocity;
         }
 
-        if (_currentTargetLocation == NPCsGlobalVariables.DefaultInitialVector ||
-            Vector3.Distance(_npcManager.transform.position, nodeWithoutY) <
-            NPCsGlobalVariables.WithinTargetNodeRange) // TODO Add check if we can see target node
+        private void ComputeAStarForSteering()
         {
-            Node npcNode = NavigationGraph.GetClosestNode(_npcManager.transform.position);
+            Node npcNode = NavigationGraph.GetClosestNode(NpcManager.transform.position);
             Node playerNode = NavigationGraph.GetClosestNode(_playerData.LastSeenPosition);
 
             List<Node> path = Pathfinding.Astar(npcNode, playerNode);
 
-            if (path.Count < 2)
+            if (path == null)
+            {
+                Debug.Log(npcNode);
+            }
+
+            _currentTargetLocation = path[NpcGlobalVariables.NextElementInPath].position;
+        }
+
+        private void ComputeCurrentTargetLocation()
+        {
+            Ray ray = new Ray(NpcManager.transform.position, _playerData.LastSeenPosition);
```

Comment:
There is an issue with the Ray
You are giving 2 positions, but the constructor of the Ray expect the 2nd parameter to be a **direction**
![image](https://user-images.githubusercontent.com/33881730/115955537-0be4f900-a4c5-11eb-9032-d4fe80c7fc8d.png)

Replace it by smth like `_playerData.LastSeenPosition - NpcManager.transform.position` or some `Vector3` equivalent to get a direction from a start and a end Vector

Reply:
On it